### DISCUSSION TITLE
custom shader using "version 300 es" throws error

### DISCUSSION THREAD

**Discussion by gallickgunner** [visgl/deck.gl#5565] [Q&A] [answered]
Hello. From what I've gathered deckgl is using webgl2 so OpenGL ES Shading language 3.0 should be supported. When I try to add a version directive "version 300 es" at the very top of the vertex shader (the first line), the layer doesn't get rendered and I get the warning 

`WebGL - Invalid Operation : useProgram: program not valid`

Removing the version directive correctly renders the layer without any error. I need glsl es 3.0 for bitwise operators inside the shader. Btw, I'm using chrome v88 and webgl2 is supported (did an online test as well)
Thanks.

**Comment by Pessimistress** [comment, marked as answer]
Hard to tell what's going on since you are not posting the shader compilation error. The [SimpleMeshLayer](https://github.com/visgl/deck.gl/tree/master/modules/mesh-layers/src/simple-mesh-layer) uses glsl es 3.0.

**Reply by gallickgunner** [reply]
I actually don't get any, that's the problem. Just the message I posted. Get that quite a few times around 250 or so at the start.

If I remove the directive, then I get shader errors regarding the bitwise operators. After putting the version directive, I don't get any error just a yellow colored message stating the program isn't valid.

**Reply by Pessimistress** [reply]
Are you adding the directive to your fragment shader also? WebGL doesn't allow mix and match of different versions.

**Reply by gallickgunner** [reply]
Dang. That was it. I forgot to add it in fragment shader. The error disappears but I still don't see the arc layer. I wonder what's up. I have nothing changed in the shader code though, just added the directive. I'll see if I can find out what's causing the problem. Thanks for the tip tho.

**Reply by gallickgunner** [reply]
Ok got it. Thanks for the help again. I was basically setting color `gl_Fragcolor` instead of defining an out variable like the simple mesh layer fragment shader does.

**Reply by jeffpeck10x** [reply]
aha! This was helpful!
